dashboard/modules/log/log_head.py

```diff
@@ -100,3 +106,137 @@ async def run(self, server):
+        )
+        filters = [] if filters == [""] else filters
+
+        def contains_all_filters(log_file_name):
+            return all(f in log_file_name for f in filters)
+
+        filtered = list(filter(contains_all_filters, reply.log_files))
+        logs = {}
+        logs["worker_errors"] = list(
+            filter(lambda s: "worker" in s and s.endswith(".err"), filtered)
+        )
+        logs["worker_outs"] = list(
+            filter(lambda s: "worker" in s and s.endswith(".out"), filtered)
+        )
+        for lang in ray_constants.LANGUAGE_WORKER_TYPES:
+            logs[f"{lang}_core_worker_logs"] = list(
+                filter(
+                    lambda s: f"{lang}-core-worker" in s and s.endswith(".log"),
+                    filtered,
+                )
+            )
+            logs[f"{lang}_driver_logs"] = list(
+                filter(
+                    lambda s: f"{lang}-core-driver" in s and s.endswith(".log"),
+                    filtered,
+                )
+            )
+        logs["dashboard"] = list(filter(lambda s: "dashboard" in s, filtered))
+        logs["raylet"] = list(filter(lambda s: "raylet" in s, filtered))
+        logs["gcs_server"] = list(filter(lambda s: "gcs" in s, filtered))
+        logs["ray_client"] = list(filter(lambda s: "ray_client" in s, filtered))
+        logs["autoscaler"] = list(
+            filter(lambda s: "monitor" in s and "log_monitor" not in s, filtered)
+        )
+        logs["runtime_env"] = list(filter(lambda s: "runtime_env" in s, filtered))
+        logs["folders"] = list(filter(lambda s: "." not in s, filtered))
+        logs["misc"] = list(
+            filter(lambda s: all([s not in logs[k] for k in logs]), filtered)
+        )
+        return logs
+
+    async def wait_until_initialized(self):
+        POLL_SLEEP_TIME = 0.5
+        POLL_RETRIES = 10
+        for _ in range(POLL_RETRIES):
+            if self._stubs != {}:
+                return None
+            await asyncio.sleep(POLL_SLEEP_TIME)
+        return aiohttp.web.HTTPGatewayTimeout(
+            reason="Could not connect to agents via gRPC after "
+            f"{POLL_SLEEP_TIME * POLL_RETRIES} seconds."
+        )
+
+    async def list_logs(self, node_id_query: str, filters: [str]):
+        """
+        Helper function to list the logs by querying each agent
+        on each cluster via gRPC.
+        """
+        response = {}
+        tasks = []
+        for node_id, grpc_stub in self._stubs.items():
+            if node_id_query is None or node_id_query == node_id:
+
+                async def coro():
+                    response[node_id] = await self.get_logs_json_index(
+                        grpc_stub, filters
+                    )
+
+                tasks.append(coro())
+        await asyncio.gather(*tasks)
+        return response
+        
+    @routes.get("/api/experimental/logs/list")
+    async def handle_list_logs(self, req):
```

Comment:
```suggestion
    async def list_logs(self, req):
```
I think the helper function could be `_list_logs`? I am fine with handle_list_logs, so it is up to you

Reply:
yep I think _XXX will be for all internal methods. I think handle is a good way to indicate HTTP handler.

Reply:
The thing is we don't follow this anywhere. Imo, doing only partially is worse than not doing it. But it is minor, so I will leave it up to you!